Module4.3_Search_and_Sorting_Algorithms/activities/Act1_SearchingAndSorting/1.md

```diff
@@ -6,15 +6,31 @@
 
 **The Importance of Searching**
 
-Searching for elements in a data structure is a key concept in computer science. The applications are endless! 
+Searching is the process of finding an element in a data structure, and in computer science, there are many different kinds of sorting, such as linear search, binary search, depth-first search, breadth-first search, and so on. We will be explaining these searching methods in more depth in this activity, and we will start with breadth-first search below! Searching is a key concept in computer science—the applications are endless! 
 
 Suppose, for example, I love watching movies. I have collections of all of the *Harry Potter*, *The Lord of the Rings*, and *James Bond* films. I have invited some friends to binge-watch them with me tomorrow morning, but I woke up late and need to find the movies quickly! How can I find the movies that I want to watch in the most efficient manner possible, before the doorbell rings?
 
 ![](https://images.pexels.com/photos/1005012/pexels-photo-1005012.jpeg?cs=srgb&dl=harry-potter-book-and-black-headphones-with-trinket-1005012.jpg&fm=jpg)   
 
-Now extend this problem to wide-scale computer systems. I need to find certain information (say, for example, medical records) from my stored data in a timely fashion. What sort of algorithms should I implement? Luckily, Computer Scientists have developed answers for us!
+Now extend this problem to wide-scale computer systems. What if we needed to find certain information (say, for example, medical records) from stored data in a timely fashion? What sort of algorithms should we implement? Luckily, Computer Scientists have developed answers for us!
 
-**Breadth-first search** (**BFS**) is an algorithm for traversing or searching tree or graph data structures. It starts at the tree root (or some arbitrary node of a graph, sometimes referred to as a 'search key'), and explores all of the neighbor nodes at the present depth prior to moving on to the nodes at the next depth level.
+
+
+**The Importance of Graphs**
+
+A graph in computer science is a visual representation of a set of nodes that are connected together by edges, and graphs allow us to view information and relationships in data structures in a way that is easier to read and understand!
```

Comment:
Explanation of nodes and edges is good initial start but needs a more technical definition to supplement this introduction.